File: Example/Objective-C/KGViewSeparatorsExample/ViewController.m
//
//  ViewController.m
//  KGViewSeparators
//
//  Created by Krisjanis Gaidis on 12/27/15.
//  Copyright © 2015 KG. All rights reserved.
//

#import "ViewController.h"
#import "UITableViewCell+Separators.h"

static const NSInteger kNumberOfRows = 100;
static const CGFloat kCellHeight = 44.0;

@interface ViewController () <UITableViewDataSource, UITableViewDelegate>

// IBOutlet's
@property (weak, nonatomic) IBOutlet UITableView *tableView;

@end

@implementation ViewController

#pragma mark - Lifecycle -

- (void)viewDidLoad {
    [super viewDidLoad];
    self.tableView.dataSource = self;
    self.tableView.delegate = self;
    self.tableView.separatorStyle = UITableViewCellSeparatorStyleNone;
}

#pragma mark - Overrides -

- (BOOL)prefersStatusBarHidden {
    return true;
}

#pragma mark - <UITableViewDataSource/UITableViewDelegate> -

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return kNumberOfRows;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {

    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"Cell" forIndexPath:indexPath];

    BOOL even = (indexPath.row % 2 == 0);
    [cell showGreenTopSeparator:even];
    [cell showRedBottomSeparator:even];
    
    if (indexPath.row % 4 == 0) {
        [cell showBlackBottomSeparator:true];
    }

    return cell;
}

- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath {
    return kCellHeight;
}

@end

File: Example/Swift/KGViewSeparatorsExample/ViewController.swift
//
//  ViewController.swift
//  KGViewSeparatorsExample
//
//  Created by Krisjanis Gaidis on 12/28/15.
//  Copyright © 2015 KG. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    // MARK: - Constants -
    private let kNumberOfRows = 100
    private let kCellHeight: CGFloat = 44.0
    
    // MARK: - IBOutlets -
    @IBOutlet private weak var tableView: UITableView!

    // MARK: - Lifecycle -
    override func viewDidLoad() {
        super.viewDidLoad()
        tableView.dataSource = self
        tableView.delegate = self
        tableView.separatorStyle = .None
    }
    
    // MARK: - Overrides -
    override func prefersStatusBarHidden() -> Bool {
        return true
    }
}

// MARK: - <UITableViewDataSource, UITableViewDelegate> -
extension ViewController: UITableViewDataSource, UITableViewDelegate {
    
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return kNumberOfRows
    }
    
    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        
        let cell = tableView.dequeueReusableCellWithIdentifier("Cell", forIndexPath: indexPath)

        cell.showTopSeparator(indexPath.row == 0)
        cell.showBottomSeparator(true)

        return cell
    }
    
    func tableView(tableView: UITableView, heightForRowAtIndexPath indexPath: NSIndexPath) -> CGFloat {
        return kCellHeight
    }
}